C/c4Database.cc

```diff
@@ -116,18 +117,20 @@ C4Database* c4db_openNamed(C4String name,
         return nullptr;
     FilePath path = dbPath(name, config->parentDirectory);
     C4DatabaseConfig oldConfig = newToOldConfig(config);
-    return c4db_open(slice(path), &oldConfig, outError);
+    return tryCatch<C4Database*>(outError, [=] {
+        return retain(new C4Database(path, oldConfig));
+    });
 }
 
 
 C4Database* c4db_openAgain(C4Database* db,
                            C4Error *outError) noexcept
 {
-    string path = db->path();
-    return c4db_open({path.data(), path.size()}, c4db_getConfig(db), outError);
+    return c4db_openNamed(c4db_getName(db), c4db_getConfig2(db), outError);
```

Comment:
Will this still work if the db here was opened with the deprecated method?

Reply:
Yeah, because c4db_getName and c4db_getConfig2 return the right thing, no matter how the db was opened.